iceoryx_dds/test/moduletests/test_iox_to_dds.cpp

```diff
@@ -39,12 +39,12 @@ using ::testing::Return;
 using ::testing::SetArgPointee;
 
 // ======================================== Helpers ======================================== //
-using TestChannel = iox::gw::Channel<MockSubscriber<void>, MockDataWriter>;
+using TestChannel = iox::gw::Channel<MockSubscriber, MockDataWriter>;
```

Comment:
Completely unrelated thought - I think this `Channel` abstraction I introduced doesn't bring much extra value and its purpose would be easily fulfilled with something more generic like `std::pair`, if we ever end up porting this.

Reply:
Good point. Shall this be added to #376?